File: internal/constants/hooks.go
package constants

var (
	ApplypatchMsgHook        = "applypatch-msg"
	PreApplypatchHook        = "pre-applypatch"
	PostApplypatchHook       = "post-applypatch"
	PreCommitHook            = "pre-commit"
	PreMergeCommitHook       = "pre-merge-commit"
	PrepareCommitMsgHook     = "prepare-commit-msg"
	CommitMsgHook            = "commit-msg"
	PostCommitHook           = "post-commit"
	PreRebaseHook            = "pre-rebase"
	PostCheckoutHook         = "post-checkout"
	PostMergeHook            = "post-merge"
	PrePushHook              = "pre-push"
	PreReceiveHook           = "pre-receive"
	UpdateHook               = "update"
	ProcReceiveHook          = "proc-receive"
	PostReceiveHook          = "post-receive"
	PostUpdateHook           = "post-update"
	ReferenceTransactionHook = "reference-transaction"
	PushToCheckoutHook       = "push-to-checkout"
	PreAutoGcHook            = "pre-auto-gc"
	PostRewriteHook          = "post-rewrite"
	SendemailValidateHook    = "sendemail-validate"
	FsmonitorWatchmanHook    = "fsmonitor-watchman"
	P4ChangelistHook         = "p4-changelist"
	P4PrepareChangelistHook  = "p4-prepare-changelist"
	P4PostChangelistHook     = "p4-post-changelist"
	P4PreSubmitHook          = "p4-pre-submit"
	PostIndexChangeHook      = "post-index-change"
)

// HooksNames is list on supported hooks.
var HooksNames = []string{
	ApplypatchMsgHook,
	PreApplypatchHook,
	PostApplypatchHook,
	PreCommitHook,
	PreMergeCommitHook,
	PrepareCommitMsgHook,
	CommitMsgHook,
	PostCommitHook,
	PreRebaseHook,
	PostCheckoutHook,
	PostMergeHook,
	PrePushHook,
	PreReceiveHook,
	UpdateHook,
	ProcReceiveHook,
	PostReceiveHook,
	PostUpdateHook,
	ReferenceTransactionHook,
	PushToCheckoutHook,
	PreAutoGcHook,
	PostRewriteHook,
	SendemailValidateHook,
	FsmonitorWatchmanHook,
	P4ChangelistHook,
	P4PrepareChangelistHook,
	P4PostChangelistHook,
	P4PreSubmitHook,
	PostIndexChangeHook,
}

File: tests/common/hooks.rs
#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
pub enum Hook {
    ApplypatchMsg,
    PreApplypatch,
    PostApplypatch,
    PreCommit,
    PreMergeCommit,
    PrepareCommitMsg,
    CommitMsg,
    PostCommit,
    PreRebase,
    PostCheckout,
    PostMerge,
    PrePush,
    PreReceive,
    Update,
    ProcReceive,
    PostReceive,
    PostUpdate,
    ReferenceTransaction,
    PushToCheckout,
    PreAutoGc,
    PostRewrite,
    SendemailValidate,
    FsmonitorWatchman,
    P4Changelist,
    P4PrepareChangelist,
    P4PostChangelist,
    P4PreSubmit,
    PostIndexChange,
}

impl Hook {
    /// Возвращает срез со всеми вариантами enum
    pub fn iter() -> &'static [Hook] {
        &[
            Hook::ApplypatchMsg,
            Hook::PreApplypatch,
            Hook::PostApplypatch,
            Hook::PreCommit,
            Hook::PreMergeCommit,
            Hook::PrepareCommitMsg,
            Hook::CommitMsg,
            Hook::PostCommit,
            Hook::PreRebase,
            Hook::PostCheckout,
            Hook::PostMerge,
            Hook::PrePush,
            Hook::PreReceive,
            Hook::Update,
            Hook::ProcReceive,
            Hook::PostReceive,
            Hook::PostUpdate,
            Hook::ReferenceTransaction,
            Hook::PushToCheckout,
            Hook::PreAutoGc,
            Hook::PostRewrite,
            Hook::SendemailValidate,
            Hook::FsmonitorWatchman,
            Hook::P4Changelist,
            Hook::P4PrepareChangelist,
            Hook::P4PostChangelist,
            Hook::P4PreSubmit,
            Hook::PostIndexChange,
        ]
    }

    /// Возвращает строковое значение, аналог serde(rename)
    pub fn as_str(&self) -> &'static str {
        match self {
            Hook::ApplypatchMsg => "applypatch-msg",
            Hook::PreApplypatch => "pre-applypatch",
            Hook::PostApplypatch => "post-applypatch",
            Hook::PreCommit => "pre-commit",
            Hook::PreMergeCommit => "pre-merge-commit",
            Hook::PrepareCommitMsg => "prepare-commit-msg",
            Hook::CommitMsg => "commit-msg",
            Hook::PostCommit => "post-commit",
            Hook::PreRebase => "pre-rebase",
            Hook::PostCheckout => "post-checkout",
            Hook::PostMerge => "post-merge",
            Hook::PrePush => "pre-push",
            Hook::PreReceive => "pre-receive",
            Hook::Update => "update",
            Hook::ProcReceive => "proc-receive",
            Hook::PostReceive => "post-receive",
            Hook::PostUpdate => "post-update",
            Hook::ReferenceTransaction => "reference-transaction",
            Hook::PushToCheckout => "push-to-checkout",
            Hook::PreAutoGc => "pre-auto-gc",
            Hook::PostRewrite => "post-rewrite",
            Hook::SendemailValidate => "sendemail-validate",
            Hook::FsmonitorWatchman => "fsmonitor-watchman",
            Hook::P4Changelist => "p4-changelist",
            Hook::P4PrepareChangelist => "p4-prepare-changelist",
            Hook::P4PostChangelist => "p4-post-changelist",
            Hook::P4PreSubmit => "p4-pre-submit",
            Hook::PostIndexChange => "post-index-change",
        }
    }
}

impl std::fmt::Display for Hook {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        write!(f, "{}", self.as_str())
    }
}